chemprop/nn/predictors.py

```diff
@@ -114,8 +114,11 @@ def forward(self, Z: Tensor) -> Tensor:
     def train_step(self, Z: Tensor) -> Tensor:
         return self.ffn(Z)
 
-    def encode(self, Z: Tensor) -> Tensor:
-        return self.ffn[:-1](Z)
+    def encode(self, Z: Tensor, index: int) -> Tensor:
+        n_layers = len(self.ffn) - 1
+        if index > n_layers:
+            raise ValueError(f"The index should not be larger then the `n_layers` of {n_layers}.")
+        return self.ffn[:index](Z)
```

Comment:
```suggestion
        return self.ffn[:index](Z)
```
There's no need to error check as all integer values are valid here based on the code. We should, however, document the implication of these values.